apiserver/facades/agent/uniter/networkinfo.go

```diff
@@ -76,10 +76,26 @@ func NewNetworkInfoForStrategy(
 		return nil, errors.Trace(err)
 	}
 
+	// Initialise the bindings map with all application endpoints.
+	// This will include those for which there is no explicit binding,
+	// such as the juju-info endpoint.
+	endpoints, err := app.Endpoints()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	allBindings := make(map[string]string)
+	for _, ep := range endpoints {
+		allBindings[ep.Name] = corenetwork.AlphaSpaceId
```

Comment:
shouldn't the space be the app/model default and not the alpha space? the machine may not be in the alpha space.

Reply:
Arguably, yes. This however preserves the prior behaviour of returning network info for the alpha space where an endpoint name not in the bound endpoints is requested.

Note that the result is:
- CAAS - everything is in the alpha space anyway.
- IAAS - still has the trapdoor of treating alpha as a space-agnostic request and so returns all it can. We still get interfaces returned.